### ISSUE TITLE
Error in doc of DatetimeIndex.indexer_between_time

### ISSUE THREAD

**Issue by joseRLC** [pydata/pandas#12038]
Hello everyone, 

This is not a major issue: I was trying to use the ```tz``` parameter as indicated on the [documentation](http://pandas.pydata.org/pandas-docs/version/0.17.1/generated/pandas.DatetimeIndex.indexer_between_time.html?highlight=indexer_between_time#pandas.DatetimeIndex.indexer_between_time) of ```DatetimeIndex.indexer_between_time``` and it turns out that ```tz``` is not implemented.

Here is an example:
```python
import pandas as pd
pd.DatetimeIndex(['2016-01-01 00:00:00', '2016-01-01 01:00:00', '2016-01-01 02:00:00']).indexer_between_time('01:00', '02:00', tz='Europe/Paris')
```
```
Traceback (most recent call last):
  File "~/anaconda/lib/python2.7/site-packages/IPython/core/interactiveshell.py", line 3066, in run_code
    exec(code_obj, self.user_global_ns, self.user_ns)
  File "<ipython-input-98-258497b42264>", line 1, in <module>
    pd.DatetimeIndex(['2016-01-01 00:00:00', '2016-01-01 01:00:00', '2016-01-01 02:00:00']).indexer_between_time('01:00', '02:00', tz='Europe/Paris')
TypeError: indexer_between_time() got an unexpected keyword argument 'tz'
```

**Comment by rockg** [comment]
That doc should be fixed (and you see the signature no longer has a tz argument).  Having a `tz` makes no sense in this context.

**Comment by jreback** [comment]
@joseRLC want to do a pull-request for that?

**Comment by RahulHP** [comment]
Can I try this out? This seems like a good issue for a first PR.

**Comment by jorisvandenbossche** [comment]
@RahulHP Certainly, go for it! If you have any questions about the workflow (see http://pandas.pydata.org/pandas-docs/stable/contributing.html), just ask (you can also use the gitter channel for that: https://gitter.im/pydata/pandas)

**Comment by jorisvandenbossche** [comment]
Closed by #12043
